Sources/AWSLambdaRuntimeCore/LambdaRunner.swift

```diff
@@ -80,13 +89,18 @@ extension Lambda {
                         if case .failure(let error) = result {
                             logger.warning("lambda handler returned an error: \(error)")
                         }
-                        return (invocation, result)
+                        return (invocation, result, baggage)
                     }
-            }.flatMap { invocation, result in
+            }.flatMap { (invocation, result, baggage: BaggageContext) in
                 // 3. report results to runtime engine
-                self.runtimeClient.reportResults(logger: logger, invocation: invocation, result: result).peekError { error in
-                    logger.error("could not report results to lambda runtime engine: \(error)")
+                self.tracer.segment(name: "ReportResults", baggage: baggage) { _ in
+                    self.runtimeClient.reportResults(logger: logger, invocation: invocation, result: result).peekError { error in
+                        logger.error("could not report results to lambda runtime engine: \(error)")
+                    }
                 }
+            }.flatMap {
+                // flush the tracer after each invocation
+                self.tracer.flush(on: self.eventLoop)
```

Comment:
should trace deeper in this method, e.g. the getNextInvocation call?

Reply:
would this be `always` instead of `flatMap`?

Reply:
> should trace deeper in this method, e.g. the getNextInvocation call?

added recording of `getNextInvocation`, this is one is a bit tricky though:

1. we will get the X-Ray trace context in the `getNextInvocation` result
2. GET is blocking (one event at a time), without getting deeper inside of the HTTP client we dont know when data started (or ended) to be received
3. added simple instrumentation of the HTTPClient used by lambda runtime, however - because of the above - its only used when sending Result, at the moment (I am not sure if it adds much value that way)

> would this be always instead of flatMap?

correct, it should always flush (also when there were errors);
fixed and created a NIO convenience method `flush` to make it easier in other places